File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CustomToolEntry.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * A single tool class entry within a custom tool package
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CustomToolEntry.Builder.class)

public final class CustomToolEntry  {
    @Deprecated
    @java.beans.ConstructorProperties({"toolClassName", "displayName", "description", "version", "config", "inputSchema"})
    public CustomToolEntry(String toolClassName, String displayName, String description, String version, java.util.Map<String, Object> config, java.util.Map<String, Object> inputSchema) {
        super();
        this.toolClassName = toolClassName;
        this.displayName = displayName;
        this.description = description;
        this.version = version;
        this.config = config;
        this.inputSchema = inputSchema;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Python class name (e.g., WeatherTool, BashTool)
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("toolClassName")
        private String toolClassName;

                /**
         * Python class name (e.g., WeatherTool, BashTool)
         * @param toolClassName the value to set
         * @return this builder
         **/
        

        public Builder toolClassName(String toolClassName) {
        this.toolClassName = toolClassName;
        return this;
        }
            /**
     * Human-readable tool name shown to LLM
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

                /**
         * Human-readable tool name shown to LLM
         * @param displayName the value to set
         * @return this builder
         **/
        

        public Builder displayName(String displayName) {
        this.displayName = displayName;
        return this;
        }
            /**
     * Tool description for LLM tool selection
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

                /**
         * Tool description for LLM tool selection
         * @param description the value to set
         * @return this builder
         **/
        

        public Builder description(String description) {
        this.description = description;
        return this;
        }
            /**
     * Tool version
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("version")
        private String version;

                /**
         * Tool version
         * @param version the value to set
         * @return this builder
         **/
        

        public Builder version(String version) {
        this.version = version;
        return this;
        }
            /**
     * Tool-specific configuration values (supports template variables)
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("config")
        private java.util.Map<String, Object> config;

                /**
         * Tool-specific configuration values (supports template variables)
         * @param config the value to set
         * @return this builder
         **/
        

        public Builder config(java.util.Map<String, Object> config) {
        this.config = config;
        return this;
        }
            /**
     * JSON schema for tool input parameters
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
        private java.util.Map<String, Object> inputSchema;

                /**
         * JSON schema for tool input parameters
         * @param inputSchema the value to set
         * @return this builder
         **/
        

        public Builder inputSchema(java.util.Map<String, Object> inputSchema) {
        this.inputSchema = inputSchema;
        return this;
        }


        public CustomToolEntry build() {
            CustomToolEntry model = new CustomToolEntry(this.toolClassName
                    , this.displayName
                    , this.description
                    , this.version
                    , this.config
                    , this.inputSchema);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CustomToolEntry model) {
                this.toolClassName(model.getToolClassName());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.version(model.getVersion());
    this.config(model.getConfig());
    this.inputSchema(model.getInputSchema());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Python class name (e.g., WeatherTool, BashTool)
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolClassName")
    private final String toolClassName;

        /**
     * Python class name (e.g., WeatherTool, BashTool)
     * @return the value
     **/
    
    public String getToolClassName() {
        return toolClassName;
    }


        /**
     * Human-readable tool name shown to LLM
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * Human-readable tool name shown to LLM
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * Tool description for LLM tool selection
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Tool description for LLM tool selection
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * Tool version
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("version")
    private final String version;

        /**
     * Tool version
     * @return the value
     **/
    
    public String getVersion() {
        return version;
    }


        /**
     * Tool-specific configuration values (supports template variables)
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("config")
    private final java.util.Map<String, Object> config;

        /**
     * Tool-specific configuration values (supports template variables)
     * @return the value
     **/
    
    public java.util.Map<String, Object> getConfig() {
        return config;
    }


        /**
     * JSON schema for tool input parameters
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
    private final java.util.Map<String, Object> inputSchema;

        /**
     * JSON schema for tool input parameters
     * @return the value
     **/
    
    public java.util.Map<String, Object> getInputSchema() {
        return inputSchema;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CustomToolEntry(");
        sb.append("toolClassName=").append(String.valueOf(this.toolClassName));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", version=").append(String.valueOf(this.version));
        sb.append(", config=").append(String.valueOf(this.config));
        sb.append(", inputSchema=").append(String.valueOf(this.inputSchema));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CustomToolEntry)) {
            return false;
        }

        CustomToolEntry other = (CustomToolEntry) o;
        return java.util.Objects.equals(this.toolClassName, other.toolClassName) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.version, other.version) &&
            java.util.Objects.equals(this.config, other.config) &&
            java.util.Objects.equals(this.inputSchema, other.inputSchema);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.toolClassName == null ? 43 : this.toolClassName.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.version == null ? 43 : this.version.hashCode());
        result = (result * PRIME) + (this.config == null ? 43 : this.config.hashCode());
        result = (result * PRIME) + (this.inputSchema == null ? 43 : this.inputSchema.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/custom_tool_entry.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CustomToolEntry(object):
    """
    A single tool class entry within a custom tool package
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CustomToolEntry object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param tool_class_name:
            The value to assign to the tool_class_name property of this CustomToolEntry.
        :type tool_class_name: str

        :param display_name:
            The value to assign to the display_name property of this CustomToolEntry.
        :type display_name: str

        :param description:
            The value to assign to the description property of this CustomToolEntry.
        :type description: str

        :param version:
            The value to assign to the version property of this CustomToolEntry.
        :type version: str

        :param config:
            The value to assign to the config property of this CustomToolEntry.
        :type config: dict(str, object)

        :param input_schema:
            The value to assign to the input_schema property of this CustomToolEntry.
        :type input_schema: dict(str, object)

        """
        self.swagger_types = {
            'tool_class_name': 'str',
            'display_name': 'str',
            'description': 'str',
            'version': 'str',
            'config': 'dict(str, object)',
            'input_schema': 'dict(str, object)'
        }

        self.attribute_map = {
            'tool_class_name': 'toolClassName',
            'display_name': 'displayName',
            'description': 'description',
            'version': 'version',
            'config': 'config',
            'input_schema': 'inputSchema'
        }

        self._tool_class_name = None
        self._display_name = None
        self._description = None
        self._version = None
        self._config = None
        self._input_schema = None

    @property
    def tool_class_name(self):
        """
        **[Required]** Gets the tool_class_name of this CustomToolEntry.
        Python class name (e.g., WeatherTool, BashTool)


        :return: The tool_class_name of this CustomToolEntry.
        :rtype: str
        """
        return self._tool_class_name

    @tool_class_name.setter
    def tool_class_name(self, tool_class_name):
        """
        Sets the tool_class_name of this CustomToolEntry.
        Python class name (e.g., WeatherTool, BashTool)


        :param tool_class_name: The tool_class_name of this CustomToolEntry.
        :type: str
        """
        self._tool_class_name = tool_class_name

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this CustomToolEntry.
        Human-readable tool name shown to LLM


        :return: The display_name of this CustomToolEntry.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this CustomToolEntry.
        Human-readable tool name shown to LLM


        :param display_name: The display_name of this CustomToolEntry.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        Gets the description of this CustomToolEntry.
        Tool description for LLM tool selection


        :return: The description of this CustomToolEntry.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this CustomToolEntry.
        Tool description for LLM tool selection


        :param description: The description of this CustomToolEntry.
        :type: str
        """
        self._description = description

    @property
    def version(self):
        """
        Gets the version of this CustomToolEntry.
        Tool version


        :return: The version of this CustomToolEntry.
        :rtype: str
        """
        return self._version

    @version.setter
    def version(self, version):
        """
        Sets the version of this CustomToolEntry.
        Tool version


        :param version: The version of this CustomToolEntry.
        :type: str
        """
        self._version = version

    @property
    def config(self):
        """
        Gets the config of this CustomToolEntry.
        Tool-specific configuration values (supports template variables)


        :return: The config of this CustomToolEntry.
        :rtype: dict(str, object)
        """
        return self._config

    @config.setter
    def config(self, config):
        """
        Sets the config of this CustomToolEntry.
        Tool-specific configuration values (supports template variables)


        :param config: The config of this CustomToolEntry.
        :type: dict(str, object)
        """
        self._config = config

    @property
    def input_schema(self):
        """
        Gets the input_schema of this CustomToolEntry.
        JSON schema for tool input parameters


        :return: The input_schema of this CustomToolEntry.
        :rtype: dict(str, object)
        """
        return self._input_schema

    @input_schema.setter
    def input_schema(self, input_schema):
        """
        Sets the input_schema of this CustomToolEntry.
        JSON schema for tool input parameters


        :param input_schema: The input_schema of this CustomToolEntry.
        :type: dict(str, object)
        """
        self._input_schema = input_schema

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
